workflow/scripts/osemosys_global/transmission/main.py

```diff
@@ -0,0 +1,247 @@
+import pandas as pd
+import os
+
+from read import(
+    import_plexos_2015,
+    import_op_life,
+    import_line_data,
+    import_iar_base,
+    import_oar_base,
+    import_capact_base,
+    import_cap_cost_base,
+    import_fix_cost_base,
+    import_op_life_base,
+    import_max_cap_invest_base,
+    import_min_cap_invest_base,
+    import_res_cap_base,
+    import_tech_set_base
+)
+
+from constants import(
+    DF_IAR_CUSTOM_VAL,
+    DF_OAR_CUSTOM_VAL
+    )
+
+from activity import(
+    activity_transmission,
+    activity_transmission_limit,
+    capact_transmission
+    )
+
+from costs import get_transmission_costs
+
+from operational_life import set_op_life_transmission
+
+from investment_constraints import cap_investment_constraints_trn
+
+from user_defined_capacity import set_user_defined_capacity_trn
+
+from sets import create_tech_set_trn
+
+def main(
+    plexos_prop: pd.DataFrame,
+    default_op_life: pd.DataFrame,
+    line_data: pd.DataFrame,
+    interface_data: pd.DataFrame,
+    iar_base: pd.DataFrame,
+    oar_base: pd.DataFrame,
+    capact_base: pd.DataFrame,
+    cap_cost_base: pd.DataFrame,
+    fix_cost_base: pd.DataFrame,
+    op_life_base: pd.DataFrame,
+    max_cap_invest_base: pd.DataFrame,
+    min_cap_invest_base: pd.DataFrame,
+    res_cap_base: pd.DataFrame,
+    tech_set_base: pd.DataFrame,
+):
```

Comment:
same question, except with the arguments `default_op_life` and `tech_set_base`

Reply:
Amended, albeit `tech_set_base` is used following your latest changes.